src/App.tsx

```diff
@@ -1,10 +1,64 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+
+import { getTodos, USER_ID } from './api/todos';
+import { wait } from './utils/fetchClient';
+import { Todo } from './types/Todo';
+import { TodosFilter } from './types/TodosFilter';
+
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { Header } from './components/Header';
+import { Todos } from './components/Todos/Todos';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [queuedTodos, setQueuedTodos] = useState<Todo[]>([]);
+  const [todosFilter, setTodosFilter] = useState<TodosFilter>('All');
+
+  const [error, setError] = useState('');
+
+  const todosFromServer = useRef<Todo[]>([]);
+  const [completedTodosCount, setCompletedTodosCount] = useState<number>(0);
+
+  const countCompletedTodos = useCallback<() => void>(() => {
+    setCompletedTodosCount(
+      todosFromServer.current.filter(todo => !todo.completed).length,
+    );
+  }, []);
+
+  const updateTodosFromServer = useCallback(() => {
+    getTodos()
+      .then(todosRecieved => {
+        todosFromServer.current = [...todosRecieved];
+        setQueuedTodos([...todosRecieved]);
+        countCompletedTodos();
+      })
+      .catch(() => {
+        setError('Unable to load todos');
+        wait(3000).then(() => setError(''));
+      });
```

Comment:
```suggestion
    getTodos()
      .then(todosRecieved => {
        setTodos([...todosRecieved]);
      })
      .catch(() => {
        setError('Unable to load todos');
        wait(3000).then(() => setError(''));
      });
```
just move this logic to useEffect